test/python/providers/fake_provider/test_generic_backend_v2.py

```diff
@@ -112,3 +116,66 @@ def test_run(self):
 
         self.assertTrue(math.isclose(counts["00000"], 500, rel_tol=0.1))
         self.assertTrue(math.isclose(counts["01111"], 500, rel_tol=0.1))
+
+    def test_duration_defaults(self):
+        """Test that the basis gates are assigned duration defaults within expected ranges."""
+
+        basis_gates = ["cx", "id", "rz", "sx", "x", "sdg", "rxx"]
+        expected_durations = {
+            "cx": (8e-8, 9e-7),
+            "id": (3e-8, 6e-8),
+            "rz": (0.0, 0.0),
+            "sx": (3e-8, 6e-8),
+            "x": (3e-8, 6e-8),
+            "measure": (7e-7, 1.5e-6),
+            "sdg": (3e-8, 6e-8),
+            "rxx": (8e-8, 9e-7),
+        }
+        target = GenericBackendV2(num_qubits=2, basis_gates=basis_gates).target
+        for inst in target:
+            for qargs in target.qargs_for_operation_name(inst):
+                duration = target[inst][qargs].duration
+                if inst not in ["delay", "reset"]:
+                    self.assertGreaterEqual(duration, expected_durations[inst][0])
+                    self.assertLessEqual(duration, expected_durations[inst][1])
+
+    def test_dd_sequence(self):
+        """Test reasonable dynamical decoupling output (defined by generated instruction durations)."""
+
+        simple = QuantumCircuit(3)
+        simple.sx([0, 1, 2])
+        simple.cx(0, 1)
+        simple.cx(1, 2)
+        simple.cx(0, 1)
+
+        backend = GenericBackendV2(num_qubits=3, seed=9)
+        durations = backend.instruction_durations
+
+        pm = PassManager(
+            [
+                ALAPScheduleAnalysis(target=backend.target),
+                PadDynamicalDecoupling(durations, [XGate(), XGate()]),
+            ]
+        )
+
+        tqc = pm.run(simple)
+        expected = QuantumCircuit(3)
+        expected.sx(0)
+        expected.delay(duration=63, unit="dt", qarg=1)
+        expected.delay(duration=1968, unit="dt", qarg=2)
+        expected.sx([1, 2])
+        expected.cx(0, 1)
+        expected.delay(duration=620, unit="dt", qarg=0)
+        expected.cx(1, 2)
+        expected.x(0)
+        expected.delay(duration=361, unit="dt", qarg=2)
+        expected.x(2)
+        expected.delay(duration=1242, unit="dt", qarg=0)
+        expected.x(0)
+        expected.delay(duration=723, unit="dt", qarg=2)
+        expected.x(2)
+        expected.delay(duration=620, unit="dt", qarg=0)
+        expected.cx(0, 1)
+        expected.delay(duration=361, unit="dt", qarg=2)
+
+        self.assertEqual(tqc, expected)
```

Comment:
I'm not super keen on having a test asserting this as an invariant of `GenericBackendV2`.  Feels to me like it's more of a test of the DD passes and subject to their whims - I made my argument elsewhere in this PR and #11781 that I don't think any huge amounts of timing information should be an assumed part of the `GenericBackendV2` contract, but I also don't feel sufficiently strongly about it to block this PR on the basis of an (imo) over-assertive test.

Reply:
I also think it's too specific, I added the test before explicitly checking the ranges of the durations, which I think makes more sense for a unit test, so I removed it in [a1adc2e](https://github.com/Qiskit/qiskit/pull/11785/commits/a1adc2ea658a76fce8f004a6048afb6a3bea09e3).